xun/functions/graph.py

```diff
@@ -29,6 +29,11 @@ class CallNode:
     store. When a call is executed, the result is stored in the program store
     using the CallNode as key.
 
+    CallNodes are used as sentinel values during scheduling. To protect against
+    attempted modification during scheduling, copying of CallNodes is
+    disallowed. This is because the value a CallNode represents is not known
+    until execution and can therefore not be used.
```

Comment:
[rant starts]

I will never understand your no-copy explanation :D
You keep explaining it, I keep being confused :D 

Like in this explanation my first thought goes to "what the problem with value not known until execution"? OK, I'll copy the node and then resolve it the same way as original one? I see precisely no problem here.
If we include your "copy is pickling" explanation, then it doesn't work either. CallNodes can be pickled and are pickled, so what?

[rant ends]

Please see if there are some points in this rant and docs can be made even better :)

Reply:
Of course we know what they are!!!!
They exist and we know that they don't know what they are before execution, but nothing should stop us from copying this "I do no not know what I am" to another place and then just finding out during execution what it is separately for both nodes!